File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    arr = []
    answer = 0
    for i, priority in enumerate(priorities):  
        arr.append((i, priority))
        
    while arr:
        if arr[0][1] == max(arr, key=lambda x: x[1])[1]:  
            current = arr.pop(0)  
            answer += 1
            if current[0] == location:
                return answer
        else:
            arr.append(arr.pop(0))  

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 1;
        
        PriorityQueue<Integer> priorityQueue = new PriorityQueue<>(Collections.reverseOrder());
        Deque<int[]> queue = new ArrayDeque<>();
        
        for(int i=0; i<priorities.length; i++) {
            queue.offer(new int[]{i, priorities[i]});
            priorityQueue.offer(priorities[i]);
        }
        
        while(!priorityQueue.isEmpty()) {
            int[] info = queue.poll();
            
            if(info[1] == priorityQueue.peek()) {
                if(info[0] == location) {
                    return answer;
                } else {
                    answer += 1;
                    priorityQueue.poll();
                }
            } else {
                queue.offer(info);
            }
            
        }
        
        return answer;
    }
}